sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DefaultStep.java

```diff
@@ -139,6 +139,9 @@ public String getName() {
     @Override
     public Status getStatus() {
         synchronized (statusLock) {
+            if (status == Status.PENDING && getStrategy().isInterrupted()) {
+                return Status.WAITING;
```

Comment:
Should `status` be set to `WAITING` by whatever interrupted the plan? This is the only case where we don't return the "true" status. Or is it not possible to access the step from whatever that code path is?